contracts/BatchAuction.sol

```diff
@@ -38,4 +54,32 @@ contract BatchAuction is Ownable {
         numTokens++;
     }
 
+    function deposit(uint8 tokenIndex, uint amount) public onlyRegistered() {
+        require(amount != 0, "Must deposit positive amount");
+
+        address tokenAddress = tokenIdToAddressMap[tokenIndex];
+        require(tokenAddress != address(0), "Requested token is not registered");
+
+        require(
+            ERC20(tokenAddress).transferFrom(msg.sender, address(this), amount), 
+            "Unsuccessful transfer"
+        );
+        
+        // Increment depositIndex until it matches correct.
+        uint nextDepositIndex = block.number / 20;
+        while (depositIndex != nextDepositIndex) {
+            depositIndex++;
```

Comment:
please use here depositIndex = block.number/20

If there were for 60 blocks no deposits, I would prefer to not do the next 3 deposits split over 3 snarks

Reply:
Although we could easily not apply deposits that have depositHash = 0.